### ISSUE TITLE
Password generator options length not honoured when saving a new Login

### ISSUE THREAD

**Issue by jbrayton** [AgileBits/onepassword-app-extension#35]
I am calling storeLoginForURLString:... with this passwordGenerationOptions dictionary:

    NSDictionary *passwordGenerationOptions = @{
                                                AppExtensionGeneratedPasswordMinLengthKey: @(20),
                                                AppExtensionGeneratedPasswordMaxLengthKey: @(64)
                                                };

The resulting default 1Password entry has a password with twelve characters. The Length slider is set to 4.

**Comment by evanwalsh** [comment]
@radazzouz I'm seeing this issue in 1.2. It's not respecting my minimum or my maximum. Here's what my code looks like:

```objective-c
NSDictionary *passwordGenerationOptions = @{
        AppExtensionGeneratedPasswordMinLengthKey: @(8),
        AppExtensionGeneratedPasswordMaxLengthKey: @(40)
    };

    [[OnePasswordExtension sharedExtension] storeLoginForURLString:@"https://harvestapp.com"
                                                      loginDetails:newLoginDetails
                                         passwordGenerationOptions:passwordGenerationOptions
                                                 forViewController:self
                                                            sender:sender
                                                        completion:^(NSDictionary *loginDict, NSError *error) {
                                                            // HANDLE 1PASSWORD STUFF HERE
                                                        }];
```

What am I doing wrong?

**Comment by radazzouz** [comment]
Hi @evanwalsh,

Thanks for the heads up! You are right indeed! The funny part is that I discovered this very issue earlier today while working on issue #36.

This is a 1Password bug (a recent regression) and not an App Extension API bug, so you don't need to change anything in your code. I just filed an internal bug and we'll get to it shortly. I will also re-open this issue.

Please let me know if there's anything else I can do to help :wink: 

Cheers!

<font size='1' color='grey'><i>ref: OPI-2697</i></font>

**Comment by radazzouz** [comment]
In PR #217 and in 1Password 5.5 we improved the password generator options as follows

```
    NSDictionary *passwordGenerationOptions = @{
                                                // The minimum password length can be 4 or more.
                                                AppExtensionGeneratedPasswordMinLengthKey: @(8),

                                                // The maximum password length can be 50 or less.
                                                AppExtensionGeneratedPasswordMaxLengthKey: @(30),

                                                // If YES, the 1Password will guarantee that the generated password will contain at least one digit (number between 0 and 9). Passing NO will not exclude digits from the generated password.
                                                AppExtensionGeneratedPasswordRequireDigitsKey: @(YES),

                                                // If YES, the 1Password will guarantee that the generated password will contain at least one symbol (See the list bellow). Passing NO with will exclude symbols from the generated password.
                                                AppExtensionGeneratedPasswordRequireSymbolsKey: @(YES),

                                                // Here are all the symbols available in the the 1Password Password Generator:
                                                // !@#$%^&*()_-+=|[]{}'\";.,>?/~`
                                                // The string for AppExtensionGeneratedPasswordForbiddenCharactersKey should contain the symbols and characters that you wish 1Password to exclude from the generated password.

                                                AppExtensionGeneratedPasswordForbiddenCharactersKey: @"!@#$%/0lIO"
                                                };
``` 

Please make sure you take a peek at the save Login and change password capabilities.

**Comment by radazzouz** [comment]
Merged PR #217
